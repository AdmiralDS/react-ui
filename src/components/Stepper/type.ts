export type Orientation = 'horizontal' | 'vertical';
export type LineClamp = 1 | 2 | 3;
export type ProgressMode = 'percentage' | 'steps';
