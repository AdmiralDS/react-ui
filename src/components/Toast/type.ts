import type { NotificationProps } from '#src/components/Notification';

export type ID = number | string;

export interface IdentifyToast extends Omit<NotificationProps, 'id'> {
  id?: ID;
}
