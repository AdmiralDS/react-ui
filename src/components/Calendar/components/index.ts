export * from './Years';
export * from './Button';
export * from './Day';
export * from './DayNames';
export * from './Month';
export * from './Months';
export * from './Panel';
export * from './Week';
