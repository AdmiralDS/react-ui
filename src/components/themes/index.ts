export * from './light';
export * from './dark';
export type { Theme } from './common';
