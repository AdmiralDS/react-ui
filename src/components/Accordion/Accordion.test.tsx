import * as React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { LIGHT_THEME } from '#src/components/themes';

import type { AccordionProps } from '.';
import { Accordion, AccordionItem } from '.';

describe('Spinner', () => {
  const Comp = (props: AccordionProps) => (
    <ThemeProvider theme={LIGHT_THEME}>
      <Accordion {...props}>
        <AccordionItem id="AccordionItem_1" title="Первый пункт">
          Контент первого пункта
        </AccordionItem>
        <AccordionItem id="AccordionItem_2" title="Второй пункт">
          Аккордеон — это вертикальный список заголовков, которые, при нажатии, показывают контент находящийся под ними.
        </AccordionItem>
        <AccordionItem id="AccordionItem_3" title="Третий пункт">
          Будущее игр никогда не было таким вдохновляющим. Творчество в играх процветает. Новые сервисы позволяют
          находить больше игр, а также приближают игроков к любимым играм, стримерам и создателям. Облако создает
          огромную возможность для потоковой передачи игр консольного уровня, что позволит людям играть с теми игроками,
          с кем хочется и где захочется. И для многих из нас нет ничего более вдохновляющего, чем рассвет нового
          поколения консолей.
        </AccordionItem>
      </Accordion>
    </ThemeProvider>
  );
  it('should render default component', () => {
    const wrapper = render(<Comp />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with dimension m', () => {
    const wrapper = render(<Comp dimension="m" />);
    expect(wrapper).toMatchSnapshot();
  });
});
