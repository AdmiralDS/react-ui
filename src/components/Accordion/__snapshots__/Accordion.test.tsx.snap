// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render component with dimension m 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c9 {
  overflow: hidden;
  -webkit-transition: max-height 250ms cubic-bezier(0.4,0,0.2,1),visibility 250ms cubic-bezier(0.4,0,0.2,1);
  transition: max-height 250ms cubic-bezier(0.4,0,0.2,1),visibility 250ms cubic-bezier(0.4,0,0.2,1);
  max-height: 0;
  visibility: hidden;
}

.c8 {
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  margin-left: 8px;
}

.c8 *[fill^='#'] {
  fill: #626F84;
}

[data-disabled='true'] .c7 *[fill^='#'] {
  fill: #A7B0BE;
}

[data-dimension='m'] .c7 {
  width: 20px;
  height: 20px;
}

.c6 {
  position: relative;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 16px 16px 15px 16px;
}

[data-dimension='m'] .c6.c6 {
  padding: 10px 16px 9px 16px;
}

.c4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 54px;
  width: 100%;
  background: transparent;
  -webkit-tap-highlight-color: transparent;
  text-align: left;
  border: none;
  background: none;
  margin: 0;
  padding: 0;
  cursor: pointer;
  overflow: visible;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c4:hover {
  background: #F1F2F4;
}

.c4:hover *[fill^='#'] {
  fill: #0A54C2;
}

.c4:focus:before {
  position: absolute;
  content: '';
  border: 2px solid #0A54C2;
  top: -1px;
  left: 0;
  bottom: -1px;
  right: 0;
}

.c4:focus,
.c4 > .c5 {
  outline: none;
}

[data-dimension='m'] .c3 {
  min-height: 38px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c2 {
  border-bottom: 1px solid #E2E5E9;
  color: #2B313B;
}

.c2 > .c3 .c7 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c11 {
  box-sizing: border-box;
  overflow-y: auto;
  max-height: 100vh;
  padding: 4px 16px 16px 16px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

[data-dimension='m'] .c10 {
  padding: 4px 16px 10px 16px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c0 {
  position: relative;
}

.c0 > .c1:first-child {
  border-top: 1px solid #E2E5E9;
}

.c0 > .c1:first-child .c5 {
  padding: 9px 16px;
}

<body>
    <div>
      <div
        class="c0"
        data-dimension="m"
      >
        <div
          class="c1 c2"
        >
          <button
            aria-controls="accordion_content_AccordionItem_1"
            class="c3 c4"
            id="accordion_title_AccordionItem_1"
            role="button"
            type="button"
          >
            <span
              class="c5 c6"
              tabindex="-1"
            >
              Первый пункт
              <div
                aria-hidden="true"
                class="c7 c8"
              />
            </span>
          </button>
          <div
            class="c9"
          >
            <div
              aria-labelledby="accordion_title_AccordionItem_1"
              class="c10 c11"
              id="accordion_content_AccordionItem_1"
              role="region"
            >
              Контент первого пункта
            </div>
          </div>
        </div>
        <div
          class="c1 c2"
        >
          <button
            aria-controls="accordion_content_AccordionItem_2"
            class="c3 c4"
            id="accordion_title_AccordionItem_2"
            role="button"
            type="button"
          >
            <span
              class="c5 c6"
              tabindex="-1"
            >
              Второй пункт
              <div
                aria-hidden="true"
                class="c7 c8"
              />
            </span>
          </button>
          <div
            class="c9"
          >
            <div
              aria-labelledby="accordion_title_AccordionItem_2"
              class="c10 c11"
              id="accordion_content_AccordionItem_2"
              role="region"
            >
              Аккордеон — это вертикальный список заголовков, которые, при нажатии, показывают контент находящийся под ними.
            </div>
          </div>
        </div>
        <div
          class="c1 c2"
        >
          <button
            aria-controls="accordion_content_AccordionItem_3"
            class="c3 c4"
            id="accordion_title_AccordionItem_3"
            role="button"
            type="button"
          >
            <span
              class="c5 c6"
              tabindex="-1"
            >
              Третий пункт
              <div
                aria-hidden="true"
                class="c7 c8"
              />
            </span>
          </button>
          <div
            class="c9"
          >
            <div
              aria-labelledby="accordion_title_AccordionItem_3"
              class="c10 c11"
              id="accordion_content_AccordionItem_3"
              role="region"
            >
              Будущее игр никогда не было таким вдохновляющим. Творчество в играх процветает. Новые сервисы позволяют находить больше игр, а также приближают игроков к любимым играм, стримерам и создателям. Облако создает огромную возможность для потоковой передачи игр консольного уровня, что позволит людям играть с теми игроками, с кем хочется и где захочется. И для многих из нас нет ничего более вдохновляющего, чем рассвет нового поколения консолей.
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c9 {
  overflow: hidden;
  -webkit-transition: max-height 250ms cubic-bezier(0.4,0,0.2,1),visibility 250ms cubic-bezier(0.4,0,0.2,1);
  transition: max-height 250ms cubic-bezier(0.4,0,0.2,1),visibility 250ms cubic-bezier(0.4,0,0.2,1);
  max-height: 0;
  visibility: hidden;
}

.c8 {
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  margin-left: 8px;
}

.c8 *[fill^='#'] {
  fill: #626F84;
}

[data-disabled='true'] .c7 *[fill^='#'] {
  fill: #A7B0BE;
}

[data-dimension='m'] .c7 {
  width: 20px;
  height: 20px;
}

.c6 {
  position: relative;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 16px 16px 15px 16px;
}

[data-dimension='m'] .c6.c6 {
  padding: 10px 16px 9px 16px;
}

.c4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 54px;
  width: 100%;
  background: transparent;
  -webkit-tap-highlight-color: transparent;
  text-align: left;
  border: none;
  background: none;
  margin: 0;
  padding: 0;
  cursor: pointer;
  overflow: visible;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c4:hover {
  background: #F1F2F4;
}

.c4:hover *[fill^='#'] {
  fill: #0A54C2;
}

.c4:focus:before {
  position: absolute;
  content: '';
  border: 2px solid #0A54C2;
  top: -1px;
  left: 0;
  bottom: -1px;
  right: 0;
}

.c4:focus,
.c4 > .c5 {
  outline: none;
}

[data-dimension='m'] .c3 {
  min-height: 38px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c2 {
  border-bottom: 1px solid #E2E5E9;
  color: #2B313B;
}

.c2 > .c3 .c7 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c11 {
  box-sizing: border-box;
  overflow-y: auto;
  max-height: 100vh;
  padding: 4px 16px 16px 16px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

[data-dimension='m'] .c10 {
  padding: 4px 16px 10px 16px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c0 {
  position: relative;
}

.c0 > .c1:first-child {
  border-top: 1px solid #E2E5E9;
}

.c0 > .c1:first-child .c5 {
  padding: 9px 16px;
}

<div>
    <div
      class="c0"
      data-dimension="m"
    >
      <div
        class="c1 c2"
      >
        <button
          aria-controls="accordion_content_AccordionItem_1"
          class="c3 c4"
          id="accordion_title_AccordionItem_1"
          role="button"
          type="button"
        >
          <span
            class="c5 c6"
            tabindex="-1"
          >
            Первый пункт
            <div
              aria-hidden="true"
              class="c7 c8"
            />
          </span>
        </button>
        <div
          class="c9"
        >
          <div
            aria-labelledby="accordion_title_AccordionItem_1"
            class="c10 c11"
            id="accordion_content_AccordionItem_1"
            role="region"
          >
            Контент первого пункта
          </div>
        </div>
      </div>
      <div
        class="c1 c2"
      >
        <button
          aria-controls="accordion_content_AccordionItem_2"
          class="c3 c4"
          id="accordion_title_AccordionItem_2"
          role="button"
          type="button"
        >
          <span
            class="c5 c6"
            tabindex="-1"
          >
            Второй пункт
            <div
              aria-hidden="true"
              class="c7 c8"
            />
          </span>
        </button>
        <div
          class="c9"
        >
          <div
            aria-labelledby="accordion_title_AccordionItem_2"
            class="c10 c11"
            id="accordion_content_AccordionItem_2"
            role="region"
          >
            Аккордеон — это вертикальный список заголовков, которые, при нажатии, показывают контент находящийся под ними.
          </div>
        </div>
      </div>
      <div
        class="c1 c2"
      >
        <button
          aria-controls="accordion_content_AccordionItem_3"
          class="c3 c4"
          id="accordion_title_AccordionItem_3"
          role="button"
          type="button"
        >
          <span
            class="c5 c6"
            tabindex="-1"
          >
            Третий пункт
            <div
              aria-hidden="true"
              class="c7 c8"
            />
          </span>
        </button>
        <div
          class="c9"
        >
          <div
            aria-labelledby="accordion_title_AccordionItem_3"
            class="c10 c11"
            id="accordion_content_AccordionItem_3"
            role="region"
          >
            Будущее игр никогда не было таким вдохновляющим. Творчество в играх процветает. Новые сервисы позволяют находить больше игр, а также приближают игроков к любимым играм, стримерам и создателям. Облако создает огромную возможность для потоковой передачи игр консольного уровня, что позволит людям играть с теми игроками, с кем хочется и где захочется. И для многих из нас нет ничего более вдохновляющего, чем рассвет нового поколения консолей.
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Spinner should render default component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c9 {
  overflow: hidden;
  -webkit-transition: max-height 250ms cubic-bezier(0.4,0,0.2,1),visibility 250ms cubic-bezier(0.4,0,0.2,1);
  transition: max-height 250ms cubic-bezier(0.4,0,0.2,1),visibility 250ms cubic-bezier(0.4,0,0.2,1);
  max-height: 0;
  visibility: hidden;
}

.c8 {
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  margin-left: 8px;
}

.c8 *[fill^='#'] {
  fill: #626F84;
}

[data-disabled='true'] .c7 *[fill^='#'] {
  fill: #A7B0BE;
}

[data-dimension='m'] .c7 {
  width: 20px;
  height: 20px;
}

.c6 {
  position: relative;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 16px 16px 15px 16px;
}

[data-dimension='m'] .c6.c6 {
  padding: 10px 16px 9px 16px;
}

.c4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 54px;
  width: 100%;
  background: transparent;
  -webkit-tap-highlight-color: transparent;
  text-align: left;
  border: none;
  background: none;
  margin: 0;
  padding: 0;
  cursor: pointer;
  overflow: visible;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c4:hover {
  background: #F1F2F4;
}

.c4:hover *[fill^='#'] {
  fill: #0A54C2;
}

.c4:focus:before {
  position: absolute;
  content: '';
  border: 2px solid #0A54C2;
  top: -1px;
  left: 0;
  bottom: -1px;
  right: 0;
}

.c4:focus,
.c4 > .c5 {
  outline: none;
}

[data-dimension='m'] .c3 {
  min-height: 38px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c2 {
  border-bottom: 1px solid #E2E5E9;
  color: #2B313B;
}

.c2 > .c3 .c7 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c11 {
  box-sizing: border-box;
  overflow-y: auto;
  max-height: 100vh;
  padding: 4px 16px 16px 16px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

[data-dimension='m'] .c10 {
  padding: 4px 16px 10px 16px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c0 {
  position: relative;
}

.c0 > .c1:first-child {
  border-top: 1px solid #E2E5E9;
}

.c0 > .c1:first-child .c5 {
  padding: 15px 16px;
}

<body>
    <div>
      <div
        class="c0"
        data-dimension="l"
      >
        <div
          class="c1 c2"
        >
          <button
            aria-controls="accordion_content_AccordionItem_1"
            class="c3 c4"
            id="accordion_title_AccordionItem_1"
            role="button"
            type="button"
          >
            <span
              class="c5 c6"
              tabindex="-1"
            >
              Первый пункт
              <div
                aria-hidden="true"
                class="c7 c8"
              />
            </span>
          </button>
          <div
            class="c9"
          >
            <div
              aria-labelledby="accordion_title_AccordionItem_1"
              class="c10 c11"
              id="accordion_content_AccordionItem_1"
              role="region"
            >
              Контент первого пункта
            </div>
          </div>
        </div>
        <div
          class="c1 c2"
        >
          <button
            aria-controls="accordion_content_AccordionItem_2"
            class="c3 c4"
            id="accordion_title_AccordionItem_2"
            role="button"
            type="button"
          >
            <span
              class="c5 c6"
              tabindex="-1"
            >
              Второй пункт
              <div
                aria-hidden="true"
                class="c7 c8"
              />
            </span>
          </button>
          <div
            class="c9"
          >
            <div
              aria-labelledby="accordion_title_AccordionItem_2"
              class="c10 c11"
              id="accordion_content_AccordionItem_2"
              role="region"
            >
              Аккордеон — это вертикальный список заголовков, которые, при нажатии, показывают контент находящийся под ними.
            </div>
          </div>
        </div>
        <div
          class="c1 c2"
        >
          <button
            aria-controls="accordion_content_AccordionItem_3"
            class="c3 c4"
            id="accordion_title_AccordionItem_3"
            role="button"
            type="button"
          >
            <span
              class="c5 c6"
              tabindex="-1"
            >
              Третий пункт
              <div
                aria-hidden="true"
                class="c7 c8"
              />
            </span>
          </button>
          <div
            class="c9"
          >
            <div
              aria-labelledby="accordion_title_AccordionItem_3"
              class="c10 c11"
              id="accordion_content_AccordionItem_3"
              role="region"
            >
              Будущее игр никогда не было таким вдохновляющим. Творчество в играх процветает. Новые сервисы позволяют находить больше игр, а также приближают игроков к любимым играм, стримерам и создателям. Облако создает огромную возможность для потоковой передачи игр консольного уровня, что позволит людям играть с теми игроками, с кем хочется и где захочется. И для многих из нас нет ничего более вдохновляющего, чем рассвет нового поколения консолей.
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c9 {
  overflow: hidden;
  -webkit-transition: max-height 250ms cubic-bezier(0.4,0,0.2,1),visibility 250ms cubic-bezier(0.4,0,0.2,1);
  transition: max-height 250ms cubic-bezier(0.4,0,0.2,1),visibility 250ms cubic-bezier(0.4,0,0.2,1);
  max-height: 0;
  visibility: hidden;
}

.c8 {
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  margin-left: 8px;
}

.c8 *[fill^='#'] {
  fill: #626F84;
}

[data-disabled='true'] .c7 *[fill^='#'] {
  fill: #A7B0BE;
}

[data-dimension='m'] .c7 {
  width: 20px;
  height: 20px;
}

.c6 {
  position: relative;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 16px 16px 15px 16px;
}

[data-dimension='m'] .c6.c6 {
  padding: 10px 16px 9px 16px;
}

.c4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 54px;
  width: 100%;
  background: transparent;
  -webkit-tap-highlight-color: transparent;
  text-align: left;
  border: none;
  background: none;
  margin: 0;
  padding: 0;
  cursor: pointer;
  overflow: visible;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c4:hover {
  background: #F1F2F4;
}

.c4:hover *[fill^='#'] {
  fill: #0A54C2;
}

.c4:focus:before {
  position: absolute;
  content: '';
  border: 2px solid #0A54C2;
  top: -1px;
  left: 0;
  bottom: -1px;
  right: 0;
}

.c4:focus,
.c4 > .c5 {
  outline: none;
}

[data-dimension='m'] .c3 {
  min-height: 38px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c2 {
  border-bottom: 1px solid #E2E5E9;
  color: #2B313B;
}

.c2 > .c3 .c7 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c11 {
  box-sizing: border-box;
  overflow-y: auto;
  max-height: 100vh;
  padding: 4px 16px 16px 16px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

[data-dimension='m'] .c10 {
  padding: 4px 16px 10px 16px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c0 {
  position: relative;
}

.c0 > .c1:first-child {
  border-top: 1px solid #E2E5E9;
}

.c0 > .c1:first-child .c5 {
  padding: 15px 16px;
}

<div>
    <div
      class="c0"
      data-dimension="l"
    >
      <div
        class="c1 c2"
      >
        <button
          aria-controls="accordion_content_AccordionItem_1"
          class="c3 c4"
          id="accordion_title_AccordionItem_1"
          role="button"
          type="button"
        >
          <span
            class="c5 c6"
            tabindex="-1"
          >
            Первый пункт
            <div
              aria-hidden="true"
              class="c7 c8"
            />
          </span>
        </button>
        <div
          class="c9"
        >
          <div
            aria-labelledby="accordion_title_AccordionItem_1"
            class="c10 c11"
            id="accordion_content_AccordionItem_1"
            role="region"
          >
            Контент первого пункта
          </div>
        </div>
      </div>
      <div
        class="c1 c2"
      >
        <button
          aria-controls="accordion_content_AccordionItem_2"
          class="c3 c4"
          id="accordion_title_AccordionItem_2"
          role="button"
          type="button"
        >
          <span
            class="c5 c6"
            tabindex="-1"
          >
            Второй пункт
            <div
              aria-hidden="true"
              class="c7 c8"
            />
          </span>
        </button>
        <div
          class="c9"
        >
          <div
            aria-labelledby="accordion_title_AccordionItem_2"
            class="c10 c11"
            id="accordion_content_AccordionItem_2"
            role="region"
          >
            Аккордеон — это вертикальный список заголовков, которые, при нажатии, показывают контент находящийся под ними.
          </div>
        </div>
      </div>
      <div
        class="c1 c2"
      >
        <button
          aria-controls="accordion_content_AccordionItem_3"
          class="c3 c4"
          id="accordion_title_AccordionItem_3"
          role="button"
          type="button"
        >
          <span
            class="c5 c6"
            tabindex="-1"
          >
            Третий пункт
            <div
              aria-hidden="true"
              class="c7 c8"
            />
          </span>
        </button>
        <div
          class="c9"
        >
          <div
            aria-labelledby="accordion_title_AccordionItem_3"
            class="c10 c11"
            id="accordion_content_AccordionItem_3"
            role="region"
          >
            Будущее игр никогда не было таким вдохновляющим. Творчество в играх процветает. Новые сервисы позволяют находить больше игр, а также приближают игроков к любимым играм, стримерам и создателям. Облако создает огромную возможность для потоковой передачи игр консольного уровня, что позволит людям играть с теми игроками, с кем хочется и где захочется. И для многих из нас нет ничего более вдохновляющего, чем рассвет нового поколения консолей.
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
